addons/website_forum/controllers/main.py

```diff
@@ -104,7 +105,57 @@ def questions(self, forum, tag=None, page=1, filters='all', sorting=None, search
 
         domain = [('forum_id', '=', forum.id), ('parent_id', '=', False), ('state', '=', 'active')]
         if search:
-            domain += ['|', ('name', 'ilike', search), ('content', 'ilike', search)]
+            search_options = [match_obj.groupdict() for match_obj in re.finditer(r'''(?P<tag>(?:or\s|-)?\[.*?\])|(?P<word_phrase>".*?")|(?P<options>[a-z]+:\-?\d*)|(?P<other>\S+)''', search)]
+            # filter search_options value and combine same key's value
+            search_options = {key: [search.get(key) for search in search_options if search.get(key)] for key in {key for search_option in search_options for key in search_option}}
```

Comment:
why ? use dictionary method something like :
search = {}
for search_dict in search_options:
          for k, v in search_dict.items():
                      seach[k].append(v)